src/ImageSharp/Formats/WebP/LossyUtils.cs

```diff
@@ -0,0 +1,912 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Buffers.Binary;
+
+namespace SixLabors.ImageSharp.Formats.WebP
+{
+    internal static class LossyUtils
+    {
+        private static void Put16(int v, Span<byte> dst)
+        {
+            for (int j = 0; j < 16; ++j)
+            {
+                Memset(dst.Slice(j * WebPConstants.Bps), (byte)v, 0, 16);
+            }
+        }
+
+        public static void DC16(Span<byte> dst, Span<byte> yuv, int offset)
+        {
+            int dc = 16;
+            for (int j = 0; j < 16; ++j)
+            {
+                // DC += dst[-1 + j * BPS] + dst[j - BPS];
+                dc += yuv[offset - 1 + (j * WebPConstants.Bps)] + yuv[offset + j - WebPConstants.Bps];
+            }
+
+            Put16(dc >> 5, dst);
+        }
+
+        public static void TM16(Span<byte> dst, Span<byte> yuv, int offset)
+        {
+            TrueMotion(dst, yuv, offset, 16);
+        }
+
+        public static void VE16(Span<byte> dst, Span<byte> yuv, int offset)
+        {
+            // vertical
+            Span<byte> src = yuv.Slice(offset - WebPConstants.Bps, 16);
+            for (int j = 0; j < 16; ++j)
+            {
+                // memcpy(dst + j * BPS, dst - BPS, 16);
+                src.CopyTo(dst.Slice(j * WebPConstants.Bps));
+            }
+        }
+
+        public static void HE16(Span<byte> dst, Span<byte> yuv, int offset)
+        {
+            // horizontal
+            for (int j = 16; j > 0; --j)
+            {
+                // memset(dst, dst[-1], 16);
+                byte v = yuv[offset - 1];
```

Comment:
apply offset-- once before the loop and omit the "-1" here.
Instead of subtracting 1 16x this reduces that to once (while it's transparent to the addition in line 55)